module/kutrace_mod.c

```diff
@@ -253,6 +253,23 @@ inline void wrMSR(u32 msr, u64 value)
   asm volatile( "wrmsr" : : "a"(lo), "d"(hi), "c"(msr) );
 }
 
+inline bool is_intel(void) {
+  uint32_t eax = 0, ebx = 0, ecx = 0, edx = 0;
+  char vendor[13];
+  __asm__ __volatile("cpuid"
+                     : "=a"(eax), "=b"(ebx), "=c"(ecx), "=d"(edx)
+                     : "a"(0));
+  snprintf(vendor, sizeof(vendor), "%c%c%c%c%c%c%c%c%c%c%c%c", ebx & 0xff,
+           (ebx >> 8) & 0xff, (ebx >> 16) & 0xff, (ebx >> 24) & 0xff,
+           edx & 0xff, (edx >> 8) & 0xff, (edx >> 16) & 0xff,
+           (edx >> 24) & 0xff, ecx & 0xff, (ecx >> 8) & 0xff,
+           (ecx >> 16) & 0xff, (ecx >> 24) & 0xff);
+  if (strncmp(vendor, "GenuineIntel", 12) == 0) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
```

Comment:
Is this function called frequently? Looks like it's enough to do a binary comparison of the contents of registers ebx, edx and ecx with the corresponding portion of the string "GenuineIntel" expressed (coded) as int values. That would avoid the call to snprintf and strncmp, since we are throwing away the contents of vendor[] anyway. Am I missing something?